### PR TITLE
Refactor include and checks for CUDA/HIP math functions

### LINKED ISSUE #1335: Duplication of CUDA/HIP include checks?
While working on traits for my ongoing complex number implementation, I've noticed all math function implementation files for CUDA/HIP have [the same piece as "include-and-check"](https://github.com/alpaka-group/alpaka/blob/6861f7b31ef0725cb117d523b94fbe2c53b60b81/include/alpaka/math/abs/AbsUniformCudaHipBuiltIn.hpp#L16-L38). Maybe it is duplicated in some other files as well, but even among the math files it is present too many times. Is there a reason it is not separated into a special include file implementing just that once, and then included when needed?

### PR DESCRIPTION
**Pull request by sbastrakov**
Those checks were duplicated in multiple files.
Move this logic to a new header file.

Resolves #1335 .

### CODE REVIEW

File: include/alpaka/core/CudaHipMath.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/BoostPredef.hpp>
+#include <alpaka/core/UniformCudaHip.hpp>
+
```
Comment (sbastrakov) [inline include/alpaka/core/CudaHipMath.hpp:14]: Note: in the [original duplicated piece](https://github.com/alpaka-group/alpaka/blob/6861f7b31ef0725cb117d523b94fbe2c53b60b81/include/alpaka/math/abs/AbsUniformCudaHipBuiltIn.hpp#L16-L38) there was also checks and errors for backend and compiler mismatch. However, exactly the same checks are done in `alpaka/core/UniformCudaHip.hpp`, so I did not put them in this file as well.

File: include/alpaka/math/MathUniformCudaHipBuiltIn.hpp [hunk 2]
```diff
@@ -11,31 +11,6 @@
...
-#                include <hip/math_functions.h>
-#            else
-#                include <math_functions.hpp>
-#            endif
-#        endif
-#        if !BOOST_LANG_HIP
-#            error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
-#        endif
-#    endif
-
```
Comment (sbastrakov) [inline include/alpaka/math/MathUniformCudaHipBuiltIn.hpp:38]: I think this whole piece should not have been here at all. All the header files do the check, and this file by itself does not add anything. So here i just removed it

File: include/alpaka/core/CudaHipMath.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+#    include <cuda_runtime.h>
+#endif
+
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+#    if BOOST_COMP_NVCC >= BOOST_VERSION_NUMBER(9, 0, 0)
+#        include <cuda_runtime_api.h>
+#    else
```
Comment (bernhardmgruber) [inline include/alpaka/core/CudaHipMath.hpp]: Didn't we drop targetting CUDA via HIP? Because then we can remove this branch.

**Review by bernhardmgruber — CHANGES_REQUESTED** [review]

File: include/alpaka/core/CudaHipMath.hpp (hunk 1 above)
Comment (bernhardmgruber) [reply include/alpaka/core/CudaHipMath.hpp:14]: Fine for me!

File: include/alpaka/math/MathUniformCudaHipBuiltIn.hpp (hunk 2 above)
Comment (bernhardmgruber) [reply include/alpaka/math/MathUniformCudaHipBuiltIn.hpp:38]: Fine for me!

File: include/alpaka/core/CudaHipMath.hpp (hunk 3 above)
Comment (psychocoderHPC) [reply include/alpaka/core/CudaHipMath.hpp]: well spotted, yes CUDA support via hip was removed with #1338. This workaround can be removed.

Comment (sbastrakov) [reply include/alpaka/core/CudaHipMath.hpp]: Thanks! Will remove then

Comment (sbastrakov) [reply include/alpaka/core/CudaHipMath.hpp]: Wait, but what exactly should be removed? I am a bit confused about this part

Comment (bernhardmgruber) [reply include/alpaka/core/CudaHipMath.hpp]: I guess we can replace the entire block by:
```c++
#if defined(ALPAKA_ACC_GPU_HIP_ENABLED)
#    include <hip/math_functions.h>
#endif
```

Comment (sbastrakov) [reply include/alpaka/core/CudaHipMath.hpp]: Okay, so then there was not much point in moving it all into a separate header. But i guess it doesn't hurt anyways, just doesn't help as much as I originally thought.

Comment (psychocoderHPC) [reply include/alpaka/core/CudaHipMath.hpp]: There must be a branch for CUDA and one for HIP.

**Review by psychocoderHPC — APPROVED** [review]

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**